src/lib.rs

```diff
@@ -163,13 +166,113 @@ pub fn get_cat(id: String) -> Result<Cat, Error> {
     Ok(cat)
 }
 
+pub fn list_adopted_cat(
+    options: Options<CatFilter>,
+    between_dates: Option<BetweenDateTime>,
+) -> Result<Paged<AdoptedCat>, Error> {
+    use crate::queries::adopted_cat::AdoptedCatQueryVariables;
+    use cynic::http::ReqwestBlockingExt;
+    use cynic::QueryBuilder;
+    use queries::adopted_cat::AdoptedCatQuery;
+
+    let endpoint = env::var("STRAPI_ENDPOINT").context(EnvVarMissingSnafu {})?;
+
+    let filters: CatFiltersInput = options.filter.map_or_else(
+        || CatFiltersInput {
+            is_dead: Some(BooleanFilterInput {
+                eq: Some(false),
+                ..BooleanFilterInput::default()
+            }),
+            ..CatFiltersInput::default()
+        },
+        |filter| CatFiltersInput {
+            is_dead: Some(BooleanFilterInput {
+                eq: Some(false),
+                ..BooleanFilterInput::default()
+            }),
+            ..filter.into()
+        },
+    );
+    let pagination = options.pagination.unwrap_or_default();
+    let sort: Option<Vec<Option<String>>> = match options.sort {
+        empty if empty.is_empty() => None,
+        otherwise => Some(otherwise.into_iter().map(Some).collect()),
+    };
+
+    let between: Option<Vec<Option<DateTime>>> = between_dates.map(|dates| dates.into());
+    let vars = AdoptedCatQueryVariables {
+        cat: filters,
+        pagination,
+        sort,
+        between,
+    };
+
+    // stored in case needed for error message
+    let vars_str = serde_json::to_string(&vars);
+
+    let operation = AdoptedCatQuery::build(vars);
+    let query = operation.query.clone();
+    let client = get_client()?;
+    let response = client
+        .post(endpoint)
+        .run_graphql(operation)
+        .context(CynicRequestSnafu {})?;
+
+    if let Some(err) = response.errors {
+        let message = format!(
+            "Variables:\n{}\nGraphQL:\n{}\nError:\n{:?}",
+            vars_str.unwrap(),
+            query,
+            err
+        )
+        .to_string();
+
+        return Err(Error::RequestResultedInError { message });
+    }
+
+    let source_cats = response
+        .data
+        .context(MissingAttributeSnafu {})?
+        .adopted_cats
+        .context(MissingAttributeSnafu {})?;
+
+    let meta = source_cats.meta;
+
+    let adopted_cats: Result<Vec<AdoptedCat>, Error> = source_cats
+        .data
+        .into_iter()
+        .map(|cat_entity| {
+            let id = cat_entity.id.map(|id| id.into_inner());
+            cat_entity
+                .attributes
+                .context(MissingAttributeSnafu {})
+                .and_then(|adopted_cat| {
+                    let AdoptedCat {
+                        id: _,
+                        adoption_date,
+                        cat,
```

Comment:
pretty sure you can use `..` syntax here instead
```suggestion
                        adoption_date,
                        cat,
                        ..
```